src/thunderbots/software/ai/world/world.cpp

```diff
@@ -115,7 +118,18 @@ Team &World::mutableEnemyTeam()
 
 void World::updateRefboxGameState(const RefboxGameState &game_state)
 {
-    game_state_.updateRefboxGameState(game_state, ball_);
+    refbox_game_state_history.push_back(game_state);
+    // Take the consensus of the past 3 refbox messages
+    if (refbox_game_state_history.size() >= 3 &&
+        refbox_game_state_history[0] == refbox_game_state_history[1] &&
+        refbox_game_state_history[0] == refbox_game_state_history[2])
+    {
+        game_state_.updateRefboxGameState(game_state, ball_);
+    }
+    else
+    {
+        game_state_.updateRefboxGameState(game_state_.getRefboxGameState(), ball_);
```

Comment:
This really feels like we should have `updateRefboxGameState(RefboxGameState)` and `updateBallState(Ball)`

Reply:
You mean updating the ball used in the "gamestate statemaching" separately? Right now the ball and gamestate are updated separately, this code is not updating the ball.

Reply:
What is this code doing if not updating the ball in `game_state_`?

Reply:
fixed